sympy/simplify/tests/test_cse.py

```diff
@@ -534,6 +535,19 @@ def test_cse_ignore_issue_15002():
     rl = [e.subs(reversed(substs)) for e in reduced]
     assert rl == l
 
+
+def test_cse_unevaluated():
+    xp1 = UnevaluatedExpr(x + 1)
+    # This used to cause RecursionError
+    [(x0, ue)], [red] = cse([(-1 - xp1) / (1 - xp1)])
+    if ue == xp1:
+        assert red == (-1 - x0) / (1 - x0)
+    elif ue == -xp1:
+        assert red == (-1 + x0) / (1 + x0)
+    else:
+        raise ValueError("Unexpected common subexpression found")
```

Comment:
Would it be useful for this error message to display what unexpected common subexpression is found versus what is expected?

```suggestion
        msg = f'Expected common subexpression {xp1} or {-xp1}, instead got {ue}'
        raise ValueError(msg)
```